cms/djangoapps/contentstore/views/tests/test_item.py

```diff
@@ -184,9 +184,9 @@ def test_get_query_count(self, branching_factor, chapter_queries, section_querie
             self.client.get(reverse_usage_url('xblock_handler', self.populated_usage_keys['problem'][-1]))
 
     @ddt.data(
-        (1, 30),
-        (2, 32),
-        (3, 34),
+        (1, 37),
+        (2, 64),
+        (3, 121),
```

Comment:
Yes, I also thought that it is very significant increase.

Reply:
Ah, it looks like the calls _currently_ only occur in Studio, but I'd be worried about this bleeding over to the LMS.

Reply:
I don't think this will ever be needed on the LMS side, but if it is, it can be cached in that case since these values won't change between course-publishes.

Reply:
Given our conversation elsewhere in the PR, I'm curious to see how this drops based on the other refactoring.
Lets reevaluate following the changes.

Reply:
This level of query growth is not acceptable, even in Studio. A branching factor of 3 means 3 chapters, 9 sequences, and 27 verticals. We have outlier courses bordering on a branch factor of 10: 10 chapters, 100 sequences, 1000 units. I haven't looked at this PR enough to know exactly what's driving the growth rate, but if it's roughly doubling with each increment, it will make large courses with inline discussions unworkable.